zebra/zebra_fpm_netlink.c

```diff
@@ -580,4 +580,67 @@ int zfpm_netlink_encode_route(int cmd, rib_dest_t *dest, struct route_entry *re,
 	return netlink_route_info_encode(ri, in_buf, in_buf_len);
 }
 
+/*
+ * zfpm_netlink_encode_mac
+ *
+ * Create a netlink message corresponding to the given MAC.
+ *
+ * Returns the number of bytes written to the buffer. 0 or a negative
+ * value indicates an error.
+ */
+int zfpm_netlink_encode_mac(struct fpm_mac_info_t *mac, char *in_buf,
+			    size_t in_buf_len)
+{
+	char buf1[ETHER_ADDR_STRLEN];
+	size_t buf_offset;
+
+	struct {
+		struct nlmsghdr hdr;
+		struct ndmsg ndm;
+		char buf[0];
+	} *req;
+	req = (void *)in_buf;
+
+	buf_offset = ((char *)req->buf) - ((char *)req);
```

Comment:
There is a dedicated macro for this in the standard library.
`buf_offset = offsetof(struct req, buf)`

Reply:
??? please fix